samples/cpp/bytetracker_demo.cpp

```diff
@@ -0,0 +1,398 @@
+    {
+        cout << "failed to open the image sequence: " << folderPath << endl;
+        return -1;
+    }
+
+    // Load image sequence.
+    Mat frame;
+    int frameNumber = 0;
+    double fps = capture.get(CAP_PROP_FPS);
+    cout<<"my fps is"<<fps;
+    cv::ByteTracker::Params params;
+    params.frameRate = fps;
+    params.frameBuffer = 30;
+    Ptr<ByteTracker> tracker = ByteTracker::create(params);
+    while (capture.read(frame))
+    {
+        if (frame.empty())
+        {
+            cout << "Failed to read the frame." << endl;
+            break;
+        }
+
+        // Load model.
+        Net net;
+        net = readNet(NET_PATH);
+        // net.setPreferableBackend(dnn::DNN_TARGET_CPU);
+        // net.setPreferableTarget(dnn::DNN_TARGET_CUDA_FP16);
+        vector<Mat> detections; // Process the image.
+        vector<Detection> objects;
+        detections = preProcessImage(frame, net);
+        Mat img = postProcessImage(frame, detections, classList,
+                               objects);
+        Mat objectsMat = detectionToMat(objects);
+        Mat trackedObjects;
+        bool ok = false;
+        ok = tracker->update(objectsMat, trackedObjects);
+        if (ok)
+        {
+            for (int i = 0; i < trackedObjects.rows; i++)
+            {
+                int id_ = static_cast<int>(trackedObjects.at<float>(i, 6));
+                Scalar color = getColor(id_);
+                cv::Rect2f tlwh_(
+                    trackedObjects.at<float>(i, 0),
+                    trackedObjects.at<float>(i, 1),
+                    trackedObjects.at<float>(i, 2),
+                    trackedObjects.at<float>(i, 3));
+
+                rectangle(img, tlwh_, color, 2);
+                int xPoint = static_cast<int>(tlwh_.x);
+                int yPoint = static_cast<int>(tlwh_.y);
+                putText(img, to_string(id_), Point(xPoint, yPoint - 5), FONT_FACE, FONT_SCALE, RED);
+            }
+            writeTracksToFile(trackedObjects, TRACKINGS_OUTPUT_PATH, frameNumber);
+            writeDetectionsToFile(objects, DETECTIONS_OUTPUT_PATH, frameNumber);
+
+            // Put efficiency information.
+            // The function getPerfProfile returns the overall time for inference(t) and the timings for each of the layers(in layersTimes).
+            vector<double> layersTimes;
+            double freq = getTickFrequency() / 1000;
+            double t = net.getPerfProfile(layersTimes) / freq;
+            string label = format("Inference time : %.2f ms", t);
+            putText(img, label, Point(20, 40), FONT_FACE, FONT_SCALE, RED);
+            imshow("Output", img);
+            writer.write(img);
+            //waitKey(0);
+
+            if (waitKey(1) == 27)
+                break;
+
+            frameNumber++;
+        }
+    }
+    writer.release();
+    capture.release();
+    cout<<"Output video generated";
+
+    return 0;
+}
+
+vector<Mat> preProcessImage(Mat &inputImage, Net &net)
```

Comment:
It's preprocess + inference. I propose to rename the method to `forward` or `inference`